{"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["iOS"],"languages":["swift"],"tokens":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"makeAsyncIterator","kind":"identifier"},{"text":"() -> ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:Scs","text":"AsyncThrowingStream"},{"text":"<","kind":"text"},{"kind":"typeIdentifier","text":"Element","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Elementxmfp"},{"kind":"text","text":", "},{"identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/Failure","kind":"typeIdentifier","text":"Failure","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Failurea"},{"kind":"text","text":">."},{"text":"Iterator","kind":"typeIdentifier","preciseIdentifier":"s:Scs8IteratorV"}]}]},{"kind":"content","content":[{"level":2,"type":"heading","anchor":"return-value","text":"Return Value"},{"type":"paragraph","inlineContent":[{"text":"An iterator that produces elements asynchronously.","type":"text"}]}]}],"sections":[],"identifier":{"url":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/makeAsyncIterator()","interfaceLanguage":"swift"},"variants":[{"paths":["\/documentation\/flowkit\/asyncthrowingsubject\/makeasynciterator()"],"traits":[{"interfaceLanguage":"swift"}]}],"kind":"symbol","hierarchy":{"paths":[["doc:\/\/flowkit.FlowKit\/documentation\/FlowKit","doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject"]]},"schemaVersion":{"major":0,"minor":3,"patch":0},"abstract":[{"type":"text","text":"Creates and returns an asynchronous iterator for the sequence."}],"metadata":{"symbolKind":"method","role":"symbol","roleHeading":"Instance Method","modules":[{"name":"FlowKit"}],"fragments":[{"text":"func","kind":"keyword"},{"kind":"text","text":" "},{"text":"makeAsyncIterator","kind":"identifier"},{"text":"() -> ","kind":"text"},{"text":"AsyncThrowingStream","kind":"typeIdentifier","preciseIdentifier":"s:Scs"},{"text":"<","kind":"text"},{"preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Elementxmfp","text":"Element","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Failurea","kind":"typeIdentifier","text":"Failure"},{"kind":"text","text":">."},{"preciseIdentifier":"s:Scs8IteratorV","kind":"typeIdentifier","text":"Iterator"}],"externalID":"s:7FlowKit20AsyncThrowingSubjectV04makeC8IteratorScs0G0Vyxs5Error_p_GyF","title":"makeAsyncIterator()"},"references":{"doc://flowkit.FlowKit/documentation/FlowKit/AsyncThrowingSubject/makeAsyncIterator()":{"kind":"symbol","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/makeAsyncIterator()","type":"topic","title":"makeAsyncIterator()","url":"\/documentation\/flowkit\/asyncthrowingsubject\/makeasynciterator()","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"makeAsyncIterator","kind":"identifier"},{"text":"() -> ","kind":"text"},{"kind":"typeIdentifier","text":"AsyncThrowingStream","preciseIdentifier":"s:Scs"},{"kind":"text","text":"<"},{"text":"Element","kind":"typeIdentifier","preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Elementxmfp"},{"text":", ","kind":"text"},{"preciseIdentifier":"s:7FlowKit20AsyncThrowingSubjectV7Failurea","kind":"typeIdentifier","text":"Failure"},{"kind":"text","text":">."},{"text":"Iterator","kind":"typeIdentifier","preciseIdentifier":"s:Scs8IteratorV"}],"role":"symbol","abstract":[{"text":"Creates and returns an asynchronous iterator for the sequence.","type":"text"}]},"doc://flowkit.FlowKit/documentation/FlowKit/AsyncThrowingSubject":{"type":"topic","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject","title":"AsyncThrowingSubject","abstract":[{"type":"text","text":"A subject that allows for asynchronously sending values or terminating with an error."},{"type":"text","text":" "},{"type":"codeVoice","code":"AsyncThrowingSubject"},{"type":"text","text":" acts as a bridge between imperative code and asynchronous code, enabling"},{"text":" ","type":"text"},{"type":"text","text":"you to send values to an "},{"code":"AsyncSequence","type":"codeVoice"},{"text":" consumer while handling potential errors gracefully.","type":"text"}],"fragments":[{"text":"struct","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"AsyncThrowingSubject"}],"url":"\/documentation\/flowkit\/asyncthrowingsubject","kind":"symbol","role":"symbol","navigatorTitle":[{"text":"AsyncThrowingSubject","kind":"identifier"}]},"doc://flowkit.FlowKit/documentation/FlowKit/AsyncThrowingSubject/Failure":{"navigatorTitle":[{"text":"Failure","kind":"identifier"}],"role":"symbol","type":"topic","url":"\/documentation\/flowkit\/asyncthrowingsubject\/failure","kind":"symbol","abstract":[{"text":"The type of error that can terminate the sequence.","type":"text"}],"title":"AsyncThrowingSubject.Failure","fragments":[{"text":"typealias","kind":"keyword"},{"text":" ","kind":"text"},{"text":"Failure","kind":"identifier"}],"identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit\/AsyncThrowingSubject\/Failure"},"doc://flowkit.FlowKit/documentation/FlowKit":{"kind":"symbol","title":"FlowKit","identifier":"doc:\/\/flowkit.FlowKit\/documentation\/FlowKit","type":"topic","role":"collection","abstract":[],"url":"\/documentation\/flowkit"}}}